### PR TITLE
Group join enhancement and fixes

### PR DESCRIPTION
**Pull request by MattiaPrimavera**
- `sendRegistrationToken` async action: send the `registrationToken` only after we got a token from the API
- disable `joinGroup` form submit until the form `isValid`
- add some missing translations
- fix: use correct translation for registration error
- use join group endpoint

### CODE REVIEW

File: src/actions/userActions.ts [hunk 1]
```diff
@@ -52,6 +54,23 @@ export const setRegistrationToken = (
     },
 })
 
+export const sendRegistrationToken = (
+    t: any,
+    registrationToken: string
+) => async (dispatch, getState) => {
+    const { user: userState } = getState()
+    const { token, userId } = userState.user
```
Comment (HugoGresse) [inline src/actions/userActions.ts:62]: this should be replace by a selector: 
const token = getTokenSelector(getState)

File: src/actions/userActions.ts [hunk 2]
```diff
@@ -52,6 +54,23 @@ export const setRegistrationToken = (
...
+
+    try {
+        const api = new Api(token)
+        await api.addRegistrationToken(userId, registrationToken)
+        dispatch(setRegistrationToken(registrationToken))
+    } catch (e) {
+        console.log('addRegistrationToken error: ', e)
+        dispatch(showErrorMessage(t('errors.register')))
```
Comment (HugoGresse) [inline src/actions/userActions.ts:70]: we could have a generic error in french + a detail in english, so we doesn't need to manage it entirely.

File: src/components/JoinGroup/JoinGroupForm.tsx [hunk 3]
```diff
@@ -13,36 +13,41 @@ import { useSelector } from 'react-redux'
...
+    groupId: string
+    groupPassword: string
+    displayName: string
 }
 
 const JoinGroupForm = ({ onFormSubmit }) => {
     const { t } = useTranslation('join-group-form')
     const token = useSelector(selectToken)
 
+    const validateForm = (values) => {
```
Comment (HugoGresse) [inline src/components/JoinGroup/JoinGroupForm.tsx:25]: still not yup, thus needing manual validatio which is not helping reducing dev time

**Review by HugoGresse — CHANGES_REQUESTED** [review]

File: src/actions/userActions.ts (hunk 1 above)
Comment (MattiaPrimavera) [reply src/actions/userActions.ts:62]: even if I needed the state to also access the `userId` variable ?

File: src/actions/userActions.ts (hunk 2 above)
Comment (MattiaPrimavera) [reply src/actions/userActions.ts:70]: For the moment I've put both translations I think, but I'm not against your idea

File: src/components/JoinGroup/JoinGroupForm.tsx (hunk 3 above)
Comment (MattiaPrimavera) [reply src/components/JoinGroup/JoinGroupForm.tsx:25]: I copy/pasted from `AdminDashboard`, I didn't study yup yet

File: src/actions/userActions.ts (hunk 1 above)
Comment (HugoGresse) [reply src/actions/userActions.ts:63]: another selector with it so

File: src/components/JoinGroup/JoinGroupForm.tsx [hunk 4]
```diff
@@ -13,36 +13,41 @@ import { useSelector } from 'react-redux'
...
+        const errors: Partial<JoinGroupValues> = {}
+        const requiredFields = ['groupId', 'groupPassword', 'displayName']
+        for (const requiredField of requiredFields) {
+            if (!values[requiredField]) {
+                errors[requiredField] = 'Required'
```
Comment (HugoGresse) [inline src/components/JoinGroup/JoinGroupForm.tsx:30]: maybe in french?

**Review by HugoGresse — APPROVED** [review]

**Pull request merged**